### DISCUSSION TITLE
Create appimage for floorp

### DISCUSSION THREAD

**Discussion by ghost** [Floorp-Projects/Floorp#1082] [Feature Request]
Why not package floorp as an appimage?

**Comment by CutterKnife** [comment]
If they could find a maintainer, that might be a good idea too.

**Comment by i-h8-github** [comment]
I would really love an official appimage as well. Thank you.

**Comment by Samueru-sama** [comment]
I noticed that floorp releases a portable compressed build instead of an appimage.

In other words, **work is already done, what's missing is adding the appimage job to the github actions**

For example this simple script does it, it turns the portable build into an appimage, the only thing that needed to be added was the .desktop file which is hosted on the repo already.

```
#!/bin/sh
set -u
APP=floorp
SITE="Floorp-Projects/Floorp"

# CREATE DIRECTORIES
[ -n "$APP" ] && mkdir -p ./"$APP".AppDir && cd ./"$APP".AppDir || exit 1

# DOWNLOAD AND EXTRACT THE ARCHIVE
version=$(wget -q https://api.github.com/repos/"$SITE"/releases/latest -O - | sed 's/[()",{} ]/\n/g' | grep -o 'https.*linux.*x86_64.*bz2$' | head -1)
wget "$version" && tar fx ./*tar.* && rm -f ./*tar.* || exit 1

# DESKTOP ENTRY AND ICON
DESKTOP="https://raw.githubusercontent.com/Floorp-Projects/Floorp/ESR115/.github/floorp-debian.desktop"
wget $DESKTOP -O ./$APP.desktop  
cp ./floorp/browser/chrome/icons/default/default128.png floorp.png && ln -s floorp.png ./.DirIcon

# AppRun
cat >> ./AppRun << 'EOF'
#!/bin/sh
CURRENTDIR="$(readlink -f "$(dirname "$0")")"
exec "$CURRENTDIR"/floorp/floorp-bin "$@"
EOF
chmod a+x ./AppRun

# MAKE APPIMAGE
cd ..
APPIMAGETOOL=$(wget -q https://api.github.com/repos/probonopd/go-appimage/releases -O - | sed 's/[()",{} ]/\n/g' | grep -oi 'https.*continuous.*tool.*86_64.*mage$')
wget -q "$APPIMAGETOOL" -O ./appimagetool
chmod a+x ./appimagetool

# Do the thing!
ARCH=x86_64 VERSION="$(echo "$version" | awk -F"/" '{print $(NF-1)}')" ./appimagetool -s ./"$APP".AppDir
echo "All Done!"
```

The appimage is an easier to work with format and that makes it compatible with existing appimage utilities as well. (No need to be extracting the tarball and finding a .desktop and all of that).

**Comment by gunir** [comment]
I made a repo for Floorp AppImage, but I would love Floorp to release official instead:
https://github.com/gunir/Floorp-AppImage/releases
